core/types/block.go

```diff
@@ -1081,8 +1088,18 @@ func (t Termini) SubTerminiAtIndex(args ...int) common.Hash {
 	return t.subTermini[args[0]]
 }
 
-func (t *Termini) SetDomTerminus(domTerminus common.Hash) {
-	t.domTerminus = domTerminus
+func (t Termini) DomTerminiAtIndex(args ...int) common.Hash {
+	if len(args) == 0 {
+		panic("cannot access dom termini at index with the index")
+	}
+	return t.domTermini[args[0]]
+}
+
+func (t *Termini) SetDomTerminiAtIndex(val common.Hash, args ...int) {
+	if len(args) == 0 {
+		panic("index cannot be empty for the dom termini")
+	}
+	t.domTermini[args[0]] = val
```

Comment:
The intent for args is to take in a common.NodeLocation right? Can we reflect that in the function to highlight misuse?

Reply:
There is not default here, You have to be intentful when setting the domtermini